modules/library/main/src/main/java/org/geotools/data/FeatureSource.java

```diff
@@ -206,8 +206,36 @@
      *         or {@code -1} if this cannot be calculated.
      *
      * @throws IOException if there are errors getting the count
+     * @deprecated Please use {@link #count(Query)} instead
      */
-    int getCount(Query query) throws IOException;
+    @Deprecated
+    default int getCount(Query query) throws IOException {
+        return Math.toIntExact(count(query));
+    }
+
+    /**
+     * Gets the number of the features that would be returned by the given
+     * {@code Query}, taking into account any settings for max features and
+     * start index set on the {@code Query}.
+     * <p>
+     * It is possible that this method will return {@code -1} if the calculation
+     * of number of features is judged to be too costly by the implementing class.
+     * In this case, you might call <code>getFeatures(query).size()</code>
+     * instead.
+     * <p>
+     * Example use:<pre><code> int count = featureSource.getCount();
+     * if( count == -1 ){
+     *    count = featureSource.getFeatures( "typeName", count ).size();
```

Comment:
Is a cast or `Math.toIntExact` needed (assuming `count` is a long here).

```
      *  if( count == -1 ){
      *    count = Math.toIntExact( featureSource.getFeatures( "typeName", count ).size() );
```

Reply:
Yes, this is the result of a "inline method" refactor, I did not check call point by call point, the PR is most to try and get some grip on the discussion on gt-devel. The way it's going, it's not very much compatible with my timeframes, the DGGS activity is just a prototype at the moment and is going to end soon, I was hoping to get some useful API improvement out of it, on the side.